game/missiongenerator/convoygenerator.py

```diff
@@ -32,17 +33,46 @@ def generate(self) -> None:
                 self.generate_convoy(convoy)
 
     def generate_convoy(self, convoy: Convoy) -> VehicleGroup:
-        group = self._create_mixed_unit_group(
-            convoy.name,
-            convoy.route_start,
-            convoy.units,
-            convoy.player_owned,
-        )
-        group.add_waypoint(
-            convoy.route_end,
-            speed=kph(40).kph,
-            move_formation=PointAction.OnRoad,
-        )
+        if self.game.settings.perf_moving_convoys:
+            group = self._create_mixed_unit_group(
+                convoy.name,
+                convoy.route_start,
+                convoy.units,
+                convoy.player_owned,
+            )
+
+            group.add_waypoint(
+                convoy.route_end,
+                speed=kph(40).kph,
+                move_formation=PointAction.OnRoad,
+            )
+
+        else:
+            # perf_moving_convoys is disabled, so add the convoy between the route start and route end.
+            # This option aims to remove long routes for ground vehicles between control points,
+            # since the CPU load for long routes on DCS is pretty heavy.
+            frontline = FrontLine(convoy.origin, convoy.destination)
+
+            # Select a segment roughly a third of the way from the origin towards the destination
+            # so the convoy spawns between the control points but is still close enough to the
+            # origin CP to be targeted by BAI flights and within the protection umbrella of the CP.
+            # Convoy start and end waypoints are not the same, so it'll move a little
+            # before stopping and hopefully find a road to drive on.
+            convoy_segment = int(0.3 * len(frontline.segments))
+
+            group = self._create_mixed_unit_group(
+                convoy.name,
+                frontline.segments[convoy_segment].point_a,
+                convoy.units,
+                convoy.player_owned,
+            )
+
+            group.add_waypoint(
+                frontline.segments[convoy_segment].point_b,
+                speed=kph(40).kph,
+                move_formation=PointAction.OnRoad,
+            )
```

Comment:
If they move at all, what's the point of the option? It's not going to help performance (in fact I wouldn't be surprised if off-road navigation had _worse_ performance).

Reply:
I have had horrible CPU load when I've planned vehicle convoys (in manually designed missions) which have driven from Tblisi to Kutaisi (along roads). I noticed some campaigns have that exact same path between control points, such as the Multi-part Georgia campaign. I think the length of the path impacts the CPU load. My understanding is that plotting long paths for ground units with Combined Arms will lag the server, which is why it's usually disabled on multiplayer servers:
https://forums.eagle.ru/topic/279435-pathfinding-please/

Thisi is why I believe the shorter paths imrove performance, and initial testing with a friend of mine seems to confirm this.

Reply:
If the part that improves performance is just shorter distances, implementing https://github.com/dcs-liberation/dcs_liberation/issues/1002 would solve the performance problem and improve convoy behavior in game.